apps/explorer/src/ui/Radio.tsx

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { RadioGroup as HeadlessRadioGroup } from '@headlessui/react';
+import { type ReactNode } from 'react';
+
+import { type ExtractProps } from '~/ui/types';
+
+export type RadioGroupProps = ExtractProps<typeof HeadlessRadioGroup> & {
+    children: ReactNode;
+    ariaLabel: string;
+};
+
+export function RadioGroup({ ariaLabel, children, ...props }: RadioGroupProps) {
+    return (
+        <HeadlessRadioGroup {...props}>
+            <HeadlessRadioGroup.Label className="sr-only">
+                {ariaLabel}
+            </HeadlessRadioGroup.Label>
+            {children}
+        </HeadlessRadioGroup>
+    );
+}
+
+export type RadioOptionProps = ExtractProps<
+    typeof HeadlessRadioGroup.Option
+> & {
+    title?: string;
+    description?: string;
```

Comment:
I don't see designs in figma that have title + description?

Reply:
I see both options are there on the Radio, so I figure to just add both in here in case we will ever need them. I only use the `label` section though for the togglable panels here https://github.com/MystenLabs/sui/pull/10773

Reply:
We can remove it if you think it is not needed atm, and add it back in later if we do require the `description` section
cc: @Jordan-Mysten

Reply:
We should keep this aligned with the figma, not with Headless UI.

Reply:
@Jordan-Mysten updated to only use `Label`